swarm/storage/types.go

```diff
@@ -241,9 +242,44 @@ The ChunkStore interface is implemented by :
 type ChunkStore interface {
 	Put(*Chunk) // effectively there is no error even if there is an error
 	Get(Key) (*Chunk, error)
+	Validate(*Key, []byte) bool
 	Close()
 }
 
+type StoreParams struct {
+	*StoreParamsHidden `toml:"-"`
+	DbCapacity         uint64
+	CacheCapacity      uint
+	BaseKey            []byte
+}
+
+// toml marshalling fails on func type members
+type StoreParamsHidden struct {
+	Hash      SwarmHasher
+	Validator func(SwarmHash, *Key, []byte) bool `json:"-"`
+}
+
+func NewStoreParams(capacity uint64, hash SwarmHasher, basekey []byte) *StoreParams {
```

Comment:
This constructor is also confusing, and impossible to reason about unless you look at the implementation. We should just use meaningful defaults, with a way to modify them if necessary - either through other constructors, or with chaining.